### PR TITLE
Window management

### PR DESCRIPTION
**Pull request by nyan-ko**
Adds classes to handle navigation between windows.

### CODE REVIEW

**Review by SMateen21 — COMMENTED** [review]
Everything looks good to me, but I'd add some documentation to improve readability, so if we have someone who doesn't know the code but sees part of it, they can understand it this section (for the most part) without having to read the rest of it. I like the logic of showing a pane and using CardLayout instead of multiple JFrames.

**Review by meiiph — COMMENTED** [review]
Code-wise it looks good, just need to write documentation/Javadocs for everything. :)

**Review by lilyxmeng — COMMENTED** [review]
Agreed with Nancy's comment, just need to include Javadocs.

**Review by meiiph — APPROVED** [review]
👍

File: src/main/java/ui/windows/PaneDelegator.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        this.manager = manager;
+    }
+
+    /**
+     * If the user clicks a NavigationButton, the manager of the PaneDelegator will update the
+     * current window the user sees on screen.
+     * @param e The event to be processed.
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
```
Comment (SMateen21) [inline src/main/java/ui/windows/PaneDelegator.java:26]: Id add "throws IllegalArgumentException" to the method header.

**Review by SMateen21 — APPROVED** [review]
Looks good! Just address the small comment I've made before merging

File: src/main/java/ui/windows/PaneDelegator.java (hunk 1 above)
Comment (meiiph) [reply src/main/java/ui/windows/PaneDelegator.java:26]: Good idea, let me add that now.

**Review by meiiph — APPROVED** [review]
👍

**Review by SMateen21 — APPROVED** [review]
Looks good, everything has been addressed.

**Review by lilyxmeng — APPROVED** [review]

**Pull request merged**